core/field_mathinput.js

```diff
@@ -58,7 +58,7 @@ Blockly.FieldMathInput.prototype.spellcheck_ = false;
 Blockly.FieldMathInput.prototype.mathValidator = function(newText) {
 	// console.log("vali");
 	newText = String(newText);
-	var pattern = new RegExp("[^a-zA-Z0-9-+*/()%^ ]", "g");
+	var pattern = new RegExp("[^a-zA-Z0-9-+*/()%^. ]", "g");
```

Comment:
have you tested this correctly?
AFAIK . means "any char" in regex, so basically you allwo everything to be written. shouldn't it be \. ?

Reply:
Yes., It's tested very carefully. It now allows . but no , and no $ and so on...